### PR TITLE
Fix several OSX compilation and deprecation warnings.

### PR DESCRIPTION
**Pull request by beoran**
In particular Gestalt() has been replaced by NSAppKitVersionNumber.

### CODE REVIEW

**Comment by beoran** [comment]
This doesn't solve all problems, since there are harder cases where an API got deprecated and basically the functionality needs to be rewritten to use a newer API on newer OSX systems. But it's a start, so...

**Comment by beoran** [comment]
It's defined from OSX 10.1 and onwards. AFAICS Allegro supports OSX 10.4 and onwards.

File: src/macosx/hidjoy.m [hunk 1]
```diff
@@ -765,17 +765,18 @@ static bool get_joystick_active(ALLEGRO_JOYSTICK *joy_)
 
 #endif // Leopard+
 
+#ifndef NSAppKitVersionNumber10_5
+#define NSAppKitVersionNumber10_5 949
```
Comment (SiegeLord) [inline src/macosx/hidjoy.m]: Why do you need to define these manually?

File: addons/native_dialog/osx_dialog.m [hunk 2]
```diff
@@ -509,6 +515,13 @@ - (NSMenuItem *) build_menu_item:(ALLEGRO_MENU_ITEM *)aitem
 static volatile ALLEGRO_EVENT_QUEUE *queue = NULL;
 static ALLEGRO_MUTEX *mutex;
 
+/* Declare this here to avoid getting a warning. 
+* There seems to be no better way to do this...
+*/
+@interface NSApplication(AllegroOSX)
```
Comment (SiegeLord) [inline addons/native_dialog/osx_dialog.m]: Could you move this next to the osx_clear_mouse declaration, so all declarations are in the same spot?

File: addons/native_dialog/osx_dialog.m [hunk 3]
```diff
@@ -4,6 +4,12 @@
 
 #import <Cocoa/Cocoa.h>
 
+/* Normally we should include aintosx.h to get this prototype but it doesn't 
+* work. So repeat the prototype here as a stopgap measure.
```
Comment (SiegeLord) [inline addons/native_dialog/osx_dialog.m]: For these block comments, we typically add a space before the '*' from the second line on.

File: src/macosx/hidjoy.m (hunk 1 above)
Comment (beoran) [inline src/macosx/hidjoy.m]: The constants are not defined when compiled using an earlier or equal version of the NSAppKit. For example, on NSAppKit 10.5, the constant above will not be defined. On 10.10  NSAppKitVersionNumber10_10 is not  defined. At least  that is what I understood from the dev docs and forum threads I read.

File: addons/native_dialog/osx_dialog.m (hunk 2 above)
Comment (beoran) [inline addons/native_dialog/osx_dialog.m]: OK

File: addons/native_dialog/osx_dialog.m (hunk 3 above)
Comment (beoran) [inline addons/native_dialog/osx_dialog.m]: OK

File: src/macosx/hidjoy.m (hunk 1 above)
Comment (SiegeLord) [inline src/macosx/hidjoy.m]: Oh crud, that's a bummer. I guess this is fine then.

**Comment by beoran** [comment]
@SiegeLord, it should be as per your instructions now.

**Comment by beoran** [comment]
I merged this, but the macos port needs a lot more love. I'll really need some help from someone else who knows OSX better to clean it up better.

**Pull request merged**